tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -330,14 +332,31 @@ void ROOT::Experimental::Internal::RPageSource::PrepareLoadCluster(
    }
 }
 
+void ROOT::Experimental::Internal::RPageSource::UpdateLastUsedCluster(DescriptorId_t clusterId)
+{
+   if (fLastUsedCluster == clusterId)
+      return;
+
+   NTupleSize_t firstEntryIndex = GetSharedDescriptorGuard()->GetClusterDescriptor(clusterId).GetFirstEntryIndex();
+   auto itr = fPreloadedClusters.begin();
+   while ((itr != fPreloadedClusters.end()) && (itr->first < firstEntryIndex)) {
+      fPagePool.Evict(itr->second);
+      itr = fPreloadedClusters.erase(itr);
+   }
```

Comment:
This evicts clusters with lower entry numbers, which is sufficient for sequential access and jumping ahead. But for truly random access, shouldn't the code also evict clusters past the cluster bunch size?

Reply:
Possibly, I'm not sure if we need extra code for this. The random access pattern is arguably a misuse, or at least users should turn of the read-ahead window in this case.

Reply:
It wasn't clear to me if you wanted to implement it in this PR... I leave it up to you